Unicord Sun Valley

WamWooWam opened the pull request:
Shamelessly ripping off the Windows 11 codename for Unicord v2.0

Todo: list updates/fixes

WamWooWam committed:
Initial redesign work

WamWooWam committed:
remove useless submodules

WamWooWam committed:
Merge branch 'master' into redesign

WamWooWam committed:
Settings updates and phone fixes

WamWooWam committed:
small UX improvements

WamWooWam committed:
Submodule updates

WamWooWam committed:
Update build instructions now building is something you can actually do

WamWooWam committed:
aw fuck

WamWooWam committed:
Fix titlebar alignment

WamWooWam committed:
Potentially fix azure pipelines

WamWooWam committed:
Attempt 2

WamWooWam committed:
Remove useless artifact generation

WamWooWam committed:
Final tweaks for CI

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
Initial server folders work

WamWooWam committed:
include servers that aren't in folders for some reason

WamWooWam committed:
more rewrites

DeveloperWOW64 commented:
Has Unicord been abandoned? The last commit to the `master` branch was in April.

Omegion2 commented:
> Has Unicord been abandoned? The last commit to the `master` branch was in April.

i might think it was abandoned, because he was working on other project or so.

WamWooWam commented:
no lol i'm just busy. happy to accept and review contributions but i'm also a university student trying to focus on my education. it appears the rumours of my demise have been greatly exaggerated

WamWooWam committed:
Starting to clean stuff up

Lamparter reviewed (commented):
Something finally happened!
Can't wait to use this 👀

WamWooWam committed:
Update README.md

null66022 committed:
Add ja-jp localization (#89)

WamWooWam committed:
building

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
Enable Markdown Headers

userwindows10mobile committed:
Add Russian localization (#95)

* Create Readme

* Add files via upload

* Update Resources.resw

Добавил перевод

* ru-RU: fix translation

---------

Co-authored-by: Er2 <er2@dismail.de>

WamWooWam committed:
Translation fixes

WamWooWam committed:
.webp fixups, performance improvements, etc.

null66022 committed:
Improved Japanese translation (#99)

* Create test

* Add files via upload

Added translated files(5/24)

* Add files via upload

* Add files via upload

* Add files via upload

* Add files via upload

* Add files via upload

* Delete test

* Add files via upload

* Update VoiceSettingsPage.resw

* Update Resources.resw

* Update ChannelPage.resw

* Update AccountsSettingsPage.resw

Improved traslation

* Update AgeGatePage.resw

Improved translation

* Update ChannelPage.resw

Improved translation

* Update AboutSettingsPage.resw

Improved Translation

WamWooWam committed:
Fix a bunch of bugs (#104)

#96, #100, #101, #102

Open in New Window menu item not working for channels
Generally improve consistency

WamWooWam committed:
Hotfix: optimize handling of CHANNEL_UNREAD_UPDATE (#105)

* Fix a bunch of bugs

#96, #100, #101, #102

Open in New Window menu item not working for channels
Generally improve consistency

* Hotfix: optimise CHANNEL_UNREAD_UPDATE handling

* Update DSharpPlus

* Update MarkdownTextBlock.Methods.cs

Lamparter commented on .gitmodules:
Would you be able to use NuGet for these modules? 🤔

WamWooWam replied:
I already am to an extent, but a lot get built in tandem with the app so it makes sense to keep them as modules

Lamparter commented on Unicord.Universal/Assets/DiscordSplashScreen.Light.svg:
Don't you want to have a logo more unique to Unicord designed? I'd be happy to lend a hand creating a Fluent 2 version if you like 😁

WamWooWam replied:
It's something I'm looking at getting done eventually, TBD

Lamparter commented on Unicord.Universal/Unicord.Universal.csproj:
```diff
@@ -646,486 +603,495 @@
-      <Generator>MSBuild:Compile</Generator>
-    </Page>
-    <Page Include="Controls\Flyouts\DirectMessageContextFlyout.xaml">
+    <PackageReference Include="Humanizer">
```
There's such a thing as central package management (not sure if it exists in UWP though) that allows you to have all your packages and dependencies in the root directory in `Directory.Packages.props` or a similar file - might be worth taking a look at if you're interested 😄

WamWooWam replied:
Don't super see the point? Visual Studio has an okay UI for handling package versioning

Lamparter replied:
If you say so - though I think hence the NU1008 schema it is required in WinUI3. So I just thought it might be good to use going forward.

Lamparter reviewed (commented):
@WamWooWam here are a few simple suggestions, don't feel pressured to add them at all! 💖

WamWooWam committed:
Integrate 10Mica, improve title bars of CompactOverlay windows

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
Updates, fixes & cleanup

WamWooWam committed:
Update D#+ Fork (#110)

* Initial D#+ Update

* The Great ViewModeling™️

* The Great Commandening™️

* The Great Line-endening™️

* Don't error on store channels + null command

* Only build x86/x64

* Embed Rewrite pt. 1

* Embed Rewrite pt. 2

* Update DSharpPlus

* Bugfixes + components

* Final fixes + cleanup

* Bump package version

* Actually save the csproj

* Fix Release build errors

* Fixes

WamWooWam committed:
Emoji/Reaction fixes

WamWooWam committed:
Fix DMs list reordering

WamWooWam committed:
Basic support for threads/forum channels (#115)

* Initial threads/forums support

* bump version fix logger

* Make notifications go away if Unicord is open

* Include version information in screenshots

* Bump package version

* Emoji/Reaction fixes

* Fix DMs list reordering

* Update DSharpPlus

* Fix Windowing

* Fix a lot more issues with notifications

* Update Package.appxmanifest

WamWooWam committed:
Notifications settings UI

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
Update DSharpPlus

WamWooWam committed:
hCaptcha support, performance improvements

WamWooWam committed:
Reduce background process memory pressure

WamWooWam committed:
use less memory damn you

WamWooWam committed:
More memory reduction

WamWooWam committed:
Cleanup

WamWooWam committed:
Remove web login (it doesn't work)

WamWooWam committed:
More cleanup

WamWooWam committed:
Backport to RS2 (#119)

* More cleanup

* Initial RS2 Support

* Fix broken hamburger button

* Fix attachment overlay on RS2

* ChannelPage header cleanup

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
Themes 2.0 (#120)

WamWooWam committed:
Performance updates (#132)

* Try use DataTemplates where possible

* Testing

* Fix message editing with DataTemplates

* Restore functionality of search/pins pages

* Move forum thread stuff to its own files

* Cleanup MessageControl

* Restyling, fix #131, fix #130

WamWooWam committed:
Settings migration and bugfixes

WamWooWam committed:
Fix a bunch of styling issues in Sun Valley

WamWooWam committed:
bump target SDK version, small fixes

WamWooWam committed:
Update promo screenshot

WamWooWam committed:
why do you keep showing up go away

WamWooWam committed:
and never come back

WamWooWam committed:
Bug fixes, cleanup, performance improvements

WamWooWam committed:
"Remove and sort usings", also fix race condition

WamWooWam committed:
Additional cleanup + bugfixes

null66022 committed:
Update translation  (#138)

* Create test

* Add files via upload

Added translated files(5/24)

* Add files via upload

* Add files via upload

* Add files via upload

* Add files via upload

* Add files via upload

* Delete test

* Add files via upload

* Update VoiceSettingsPage.resw

* Update Resources.resw

* Update ChannelPage.resw

* Update AccountsSettingsPage.resw

Improved traslation

* Update AgeGatePage.resw

Improved translation

* Update ChannelPage.resw

Improved translation

* Update AboutSettingsPage.resw

Improved Translation

* Update SettingsPage.resaddw

Add notification tab

* Add files via upload

* Update NotificationsSettingsPage.resw

* Update PinsPage.resw

Add Erroredheader and Erroredsubheader

* Update ThemesSettingsPage.resw

AppThemeBlock translated

* Update MessagingSettingsPage.resw

* Update ThemesSettingsPage.resw

* Update MessagingSettingsPage.resw

Add descriptions of WebP and last accessed channel

WamWooWam committed:
Crash+Notifications fixes

WamWooWam committed:
Update build.yaml

WamWooWam committed:
Update WindowsCommunityToolkit

WamWooWam committed:
Update build.jaml

WamWooWam committed:
Update build.yaml

WamWooWam committed:
work damn you

WamWooWam committed:
i swear to god

WamWooWam committed:
asd

WamWooWam committed:
asdftghy

WamWooWam committed:
i _swear_ to god

WamWooWam committed:
:D

WamWooWam committed:
Update WindowsCommunityToolkit

WamWooWam committed:
i _swear to god_

WamWooWam committed:
i hope msbuild dies a long painful ideally fire-y death

WamWooWam committed:
hate this

Gargaj committed:
add a close / collapse button to the search pane

Gargaj committed:
Add support for Discord-internal links

WamWooWam committed:
Merge pull request #145 from Gargaj/close-search-pane

Gargaj committed:
update with a broader regex, make navigator service forward compatible

WamWooWam committed:
Merge pull request #147 from Gargaj/patch-2

Add support for Discord-internal links

cnshing committed:
docs: Update README.md Prerequisites

Contains explicit instructions on how to obtain Universal Windows Platform tools from the Visual Studio installer

cnshing committed:
docs: Update information from git submodules in `README.md`

Clarified information on submodules and visual studio

cnshing committed:
assets: Upload imgur launcher and batch build pics

From `README.md`

cnshing committed:
docs: Replace imgur assets with equivalent github hosted assets in `README.md`

Gargaj committed:
fix regexp being used even if it didn't match

WamWooWam committed:
Merge pull request #155 from Gargaj/patch-3

fix regexp being used even if it didn't match

GID0317 committed:
Add floating user panel pill and theme resources

Adds a Discord-style floating user panel pill to the left sidebar (DiscordPage.xaml).
Introduces theme resources for the pill (DiscordPage_UserPill_Background, DiscordPage_UserPill_CornerRadius) in SunValley, Fluent and Performance.
Adds bottom padding to DM/guild lists to prevent overlap with the pill.
Fix asymmetrical guild item view list
Improve the spacing between layout to  use 4 instead of 8

GID0317 committed:
Make bottom left and right DM/Channel list to not use rounded corners

WamWooWam committed:
Merge pull request #151 from cnshing/redesign

Clarify `README.md` setup instructions and fix imgur assets

GID0317 committed:
moved Font family to theme resource

Created DiscordPage_UserPill_SecondaryFontFamily resource:
SunValley (Windows 11): Segoe UI Variable Small
Fluent (Windows 10): Segoe UI
Performance (Windows 10): Segoe UI

GID0317 committed:
Update DiscordPage.xaml

dded xmlns:w1903 namespace and prefixed Grid.Shadow and ThemeShadow with w1903: so it only applies on Windows 10 1903+
Improve margin of profile to be more consistent
make font use Font family theme resource

WamWooWam committed:
Merge pull request #156 from GID0317/floating-user-pill

Implement Discord-style user pill, adjust sidebars and main layout spacing

GID0317 committed:
Update ChannelPageHeaderControl.xaml

- Rearranges the right button arrangement to follow discord button UI order
- Update ShowSidebarButton to use back symbol instead of hamburger symbol to reflect latest discord UI change

GID0317 committed:
Update ShowSidebarButton to use back symbol instead of hamburger

GID0317 committed:
Update SearchPage.xaml

-Added "no results" StackPanel with ":(" emoticon, header, and subheader (similar to PinsPage)
-improve margin of close button and text header for sun valley theme
- change the the search to use autosuggest box instead textbox to make it consistent across the application

GID0317 committed:
Update SearchPageViewModel.cs

- Added _hasSearched field to track if search was performed
- Added NoResults computed property (returns true only after search with zero results)
- Updated IsSearching, WaitingForIndex, and TotalMessages to notify NoResults
- Set _hasSearched = true in SearchAsync() method

GID0317 committed:
Add close button similar like search panel to Pins and member list side panel

GID0317 committed:
Added auto-close sidebar logic in TreeView_ItemInvoked() when guild is clicked in narrow mode (≤767px width)

This will make navigation between server much more easier since it will not automatically open the channel. similar like how mobile discord experience

GID0317 committed:
add EN,JP,RU string for the not found result in search panel

the text was using discord not found search

GID0317 committed:
Update the leave server text to use red color instead of yellow  to make it same as discord

GID0317 committed:
add profile and autosuggestbox to make similar to latest discord UI

Added profile row in NavigationView pane (avatar + name)
Added AutoSuggestBox search functionality
Compact mode handling (collapsed search, avatar-only profile)
Theme-aware spacing adjustments for SunValley vs Fluent/Performance
Profile name now uses GlobalName instead of discriminator

GID0317 committed:
update the profile card, adapt the depreciated windows 10 related to not shown in windows 11 (Live Tiles, Action Center Symbol), Update the statistic to using expander with copy button

Complete card redesign: vertical banner + overlapping avatar layout
Added users:AvatarControl with presence indicator
Display Name / Username ordering (no discriminator)
Red logout button
Theme-adaptive corner radius
Added UserPresence property with live messenger updates
Statistics moved to expander with "Copy" button

GID0317 committed:
Modernize the about page to use expander

Single expander with Community Toolkit like Settings Expander like-style header
App icon + name + description on left, version on right
Version text selectable, "v" prefix removed
Disclaimer section with border separator
Acknowledgements content properly formatted

GID0317 committed:
Bundled notification toggles under one expander Added OS-adaptive notification glyph (Win11 vs Win10)

Bundled notification toggles under one expander
Added OS-adaptive notification glyph (Win11 vs Win10)

GID0317 committed:
Replace the Windows Hello unavailable warning text with InfoBar

GID0317 committed:
make the restart dialog to immediately appeared when we click cancel it revert to previous selection before change

Immediate restart dialog on theme selection
Cancel reverts theme selection
Removed old "restart required" prompt

GID0317 committed:
Added expander divider between video/audio options

GID0317 committed:
Move "Learn more" moved into description to make it inline with windows 11 design

GID0317 committed:
Update AccountsSettingsModel.cs

Added UserPresence property (PresenceViewModel)
Registered for DiscordEventMessage<PresenceUpdateEventArgs> via messenger
Updates presence indicator in real-time

GID0317 committed:
removed DisplayVersion "v" prefix

GID0317 committed:
Added NotificationGlyph property (OS-aware)

GID0317 committed:
Added AccountsSettings_ProfileCard_CornerRadius: 0 and Added AccountsSettings_ProfileCard_BannerCornerRadius: 0

GID0317 committed:
Added AccountsSettings_ProfileCard_CornerRadius: 8 and Added AccountsSettings_ProfileCard_BannerCornerRadius: 8,8,0,0

GID0317 committed:
Added ExpandableContent property for expandable sections and Icon/content positioning improvements

GID0317 committed:
Layout refinements for Win11 polish

GID0317 committed:
Update AccountsSettingsPage.resw

StatisticsBlock.Title → "Statistics"
StatisticsBlock.Description → "View detailed statistics about your account"
CopyStatisticsButton.Content → "Copy"
CopyStatisticsButtonText.Text → "Copy"

GID0317 committed:
Update SettingsPage.resw

SearchBox.PlaceholderText → "Search"
EditProfileText.Text → "Edit Profiles"

GID0317 committed:
Update SecuritySettingsPage.resw

InfoBar warning text updates
"Learn more" button text

GID0317 committed:
Add dialog cancel string for theme restart dialog

GID0317 committed:
Update Description text for contrast hyperlink

GID0317 committed:
Update NotificationsSettingsPage.resw

Add a OS type related sting like in windows 11 we not specified the changed system and depreciated features but in windows 10 we displayed those

GID0317 committed:
Add similar string like en-US for the settings page

Added and updated several resource strings for Accounts, Messaging, Security, and general Settings pages. Changes include new labels and descriptions for statistics, updated button texts, improved clarity for dialog buttons, and additional accessibility and navigation options.

GID0317 committed:
Add similar string like en-US for the settings page

Added new French translations for statistics and profile editing in settings pages. Updated dialog button text and improved clarity in security settings. These changes enhance localization and user experience for French-speaking users.

GID0317 committed:
Add similar string like en-US for the settings page

Added and refined Japanese resource strings for Accounts, Notifications, Security, and general Settings pages. Changes include new descriptions for Windows 10/11, improved button labels, and additional context for statistics and profile editing.

GID0317 committed:
Add similar string like en-US for the settings page

Added and updated several Russian localization strings in AccountsSettingsPage, Dialogs, SecuritySettingsPage, and SettingsPage. Changes include new descriptions, button texts, and improved wording for clarity and consistency.

GID0317 committed:
Make controls margin inside the expander consistent

GID0317 committed:
Update the banner heigh to similar like discord heigh in the settings

GID0317 committed:
Update the margin  of collapsed mode search icon to correctly centered

GID0317 committed:
Clean up comments in SettingsPage.xaml.cs

Removed redundant and outdated comments

GID0317 committed:
Removed direct UI property assignments

GID0317 committed:
Bound PersonPicture.ProfilePicture and TextBlock.Text to ViewModel

GID0317 committed:
Added ProfileDisplayName and ProfileAvatarUrl properties

GID0317 committed:
Improve account page profile card text styles and layout

Adjusted border thickness and brush for avatar container, increased avatar margin, and updated text styles for display name, username, and email fields to use proper brush style and improved font weights.

GID0317 committed:
Add 'Copy User ID' option under more button to account page settings

GID0317 committed:
Fix indentation in AccountsSettingsPage.xaml

Gargaj committed:
add github action build step to restore legacy (22621) SDK

WamWooWam committed:
nit: move to commands instead of event handlers

WamWooWam committed:
Merge pull request #160 from Gargaj/patch-4

add github action build step to restore legacy (22621) SDK

WamWooWam committed:
@Gargaj may be the GOAT

WamWooWam committed:
Merge branch 'redesign' of https://github.com/UnicordDev/Unicord into redesign

WamWooWam committed:
specify actual project

WamWooWam committed:
fix build dates

WamWooWam committed:
Begone AnyCPU

WamWooWam committed:
chore: fix login race condition

WamWooWam committed:
fixes: icons, colours

WamWooWam committed:
Merge branch 'pr/157' into redesign

WamWooWam committed:
fix: missing resources, layout

WamWooWam committed:
fixes for RS2
